quickwit/quickwit-indexing/src/merge_policy/const_write_amplification.rs

```diff
@@ -0,0 +1,328 @@
+            operations[0].splits_as_slice().len(),
+            merge_policy.merge_factor
+        );
+    }
+
+    #[test]
+    fn test_const_write_merge_policy_merge_factor_max() {
+        let merge_policy = ConstWriteAmplificationMergePolicy::for_test();
+        let mut splits = (0..merge_policy.max_merge_factor + merge_policy.merge_factor - 1)
+            .map(|i| SplitMetadata {
+                split_id: format!("split-{i}"),
+                num_docs: 1_000,
+                num_merge_ops: 1,
+                ..Default::default()
+            })
+            .collect();
+        let operations: Vec<MergeOperation> = merge_policy.operations(&mut splits);
+        assert_eq!(operations.len(), 1);
+        assert_eq!(
+            operations[0].splits_as_slice().len(),
+            merge_policy.max_merge_factor
+        );
+    }
+
+    #[test]
+    fn test_const_write_merge_policy_older_first() {
+        let merge_policy = ConstWriteAmplificationMergePolicy::for_test();
+        let mut splits: Vec<SplitMetadata> = (0..merge_policy.max_merge_factor)
+            .map(|i| SplitMetadata {
+                split_id: format!("split-{i}"),
+                num_docs: 1_000,
+                create_timestamp: 1_664_000_000i64 + i as i64,
+                num_merge_ops: 1,
+                ..Default::default()
+            })
+            .collect();
+        splits.shuffle(&mut rand::thread_rng());
+        let operations: Vec<MergeOperation> = merge_policy.operations(&mut splits);
+        assert_eq!(operations.len(), 1);
+        assert_eq!(
+            operations[0].splits_as_slice().len(),
+            merge_policy.max_merge_factor
+        );
+        let split_ids: Vec<&str> = operations[0]
+            .splits_as_slice()
+            .iter()
+            .map(|split| split.split_id())
+            .collect();
+        assert_eq!(
+            &split_ids[..],
+            &["split-0", "split-1", "split-2", "split-3", "split-4"]
+        );
+    }
+
+    #[test]
+    fn test_const_write_merge_policy_target_num_docs() {
+        let merge_policy = ConstWriteAmplificationMergePolicy::for_test();
+        let mut splits = (0..4)
+            .map(|i| SplitMetadata {
+                split_id: format!("split-{i}"),
+                num_docs: (merge_policy.split_num_docs_target + 2) / 3,
+                num_merge_ops: 1,
+                ..Default::default()
+            })
+            .collect();
+        let operations: Vec<MergeOperation> = merge_policy.operations(&mut splits);
+        assert_eq!(operations.len(), 1);
+        assert_eq!(operations[0].splits_as_slice().len(), 3);
+    }
+
+    #[test]
+    fn test_const_write_amp_merge_policy_proptest() {
+        let merge_policy = ConstWriteAmplificationMergePolicy::for_test();
+        crate::merge_policy::tests::proptest_merge_policy(&merge_policy);
+    }
+
+    #[tokio::test]
+    async fn test_simulate_const_write_amplification_merge_policy() -> anyhow::Result<()> {
```

Comment:
Maybe in this simple test we can compute manually the final number of splits after applying all the merges and make an assert on this value.

Reply:
tweaked the simulator to return the final number of splits and added an assert